pkgs/applications/window-managers/tinywl/default.nix

```diff
@@ -4,25 +4,29 @@
 
 stdenv.mkDerivation {
   pname = "tinywl";
-  inherit (wlroots) version src;
-
-  sourceRoot = "${wlroots.src.name}/tinywl";
+  inherit (wlroots) version src patches postPatch;
 
   nativeBuildInputs = [ pkg-config wayland-scanner ];
   buildInputs = [ libxkbcommon pixman udev wayland wayland-protocols wlroots ];
 
+  buildPhase = ''
+    runHook preBuild
+    make -C tinywl
+    runHook postBuild
+  '';
```

Comment:
buildTarget should be used instead

Reply:
```
configuring
no configure script, doing nothing
building
no Makefile or custom buildPhase, doing nothing
installing
cp: -r not specified; omitting directory 'tinywl'
```

buildTarget will not change sourceRoot， but set `sourceRoot` will break patches form wlroots